src/components/Checkbox/index.tsx

```diff
@@ -12,123 +12,169 @@ import {
+    cursor: not-allowed;
+    color: ${({ theme }) => theme.colors.neutral.textDisabled};
+
+    ${StyledIcon} {
+      fill: ${({ theme }) => theme.colors.neutral.textDisabled};
     }
   }
-  &:focus + svg {
-    outline: 1px ${({ theme }) => theme.colors.neutral.backgroundHover} dotted;
+
+  ${StyledReakitCheckbox}:focus + ${StyledIcon} {
+    background-color: ${({ theme }) => theme.colors.primary.background};
+    fill: ${({ theme }) => theme.colors.primary.text};
   }
-`
 
-const StyledIcon = styled(Icon)`
-  box-sizing: content-box;
-`
+  ${StyledReakitCheckbox}[aria-invalid="true"]:focus + ${StyledIcon} {
+    background-color: ${({ theme }) => theme.colors.danger.background};
+    fill: ${({ theme }) => theme.colors.danger.text};
+  }
 
-const StyledChildrenContainer = styled('div', {
-  shouldForwardProp: prop => !['size'].includes(prop.toString()),
-})``
+  :hover[aria-disabled='false'] {
+    ${StyledReakitCheckbox} + ${StyledIcon} {
+      background-color: ${({ theme }) => theme.colors.primary.background};
+      fill: ${({ theme }) => theme.colors.primary.text};
+    }
+
+    ${StyledReakitCheckbox}[aria-invalid="true"] + ${StyledIcon} {
+      background-color: ${({ theme }) => theme.colors.danger.background};
+      fill: ${({ theme }) => theme.colors.danger.text};
+    }
+  }
+`
 
 const StyledActivityContainer = styled('div', {
   shouldForwardProp: prop => !['hasChildren'].includes(prop.toString()),
 })<{ hasChildren: boolean }>`
-  display: flex;
+  display: inline;
+  vertical-align: middle;
   margin-right: ${({ theme, hasChildren }) =>
     hasChildren ? theme.space[1] : 0};
 `
 
-const StyledError = styled.div`
-  font-size: 12px;
-  color: ${({ theme }) => theme.colors.danger.text};
-  padding: ${({ theme }) => `0 ${theme.space['0.5']}`};
-`
-
 type CheckboxProps = Omit<ReakitCheckboxProps, 'checked'> & {
   children?: ReactNode
-  valid?: boolean
   error?: string | ReactNode
   size?: number
   progress?: boolean
   disabled?: boolean
-  typographyVariant?: string
   checked?: boolean | 'indeterminate'
   className?: string
-} & Required<Pick<ReakitCheckboxProps, 'onChange'>> &
-  XStyledProps
+} & Required<Pick<ReakitCheckboxProps, 'onChange'>>
 
 const Checkbox = ({
   checked = false,
   onChange,
   onFocus,
   onBlur,
-  valid,
   error,
-  name = 'checkbox',
+  name,
   value,
   size = 24,
   children,
   progress = false,
   disabled = false,
   autoFocus = false,
-  typographyVariant = 'default',
   className,
-  ...props
 }: CheckboxProps) => {
   const hasChildren = !!children
   const checkbox = useCheckboxState({ state: checked })
-  const color = useMemo(() => {
-    if (disabled) return 'gray100'
-    if (valid === false || !!error) return 'warning'
-    if (valid === true) return 'success'
-    if (checkbox.state) return 'primary'
 
-    return 'gray300'
-  }, [disabled, valid, checkbox.state, error])
+  const computedName = useMemo(() => {
+    if (!name) return getUUID('checkbox')
```

Comment:
useUUID

Reply:
@chambo-e I want to be able to set a full name into my checkbox input without having generated ID next to it if name exists. Only when name prop is empty I generate a name with an ID. 

`useUUID` can't be used conditionally as it uses a `useMemo` into it. In order to be able to conditionally set this variable I had to use `getUUID`.